src/shared/util.js

```diff
@@ -1218,20 +1218,172 @@ function MessageHandler(sourceName, targetName, comObj) {
   this.sourceName = sourceName;
   this.targetName = targetName;
   this.comObj = comObj;
-  this.callbackIndex = 1;
+  this.callbackId = 1;
+  this.streamId = 1;
   this.postMessageTransfers = true;
-  var callbacksCapabilities = this.callbacksCapabilities = Object.create(null);
-  var ah = this.actionHandler = Object.create(null);
+  let streamSinks = this.streamSinks = Object.create(null);
+  let streamControllers = this.streamControllers = Object.create(null);
+  let callbacksCapabilities = this.callbacksCapabilities = Object.create(null);
+  let ah = this.actionHandler = Object.create(null);
+  let resolveCall = (fn, args, thisArg = null) => {
```

Comment:
It doesn't look like this needs to be an arrow function (since there's no `this` reference in the function), so please just do `function resolveCall(fn, args, thisArg = null) {` instead.
Also, for readability, please add a newline before this function definition.